ffi/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! Structures, macros related to FFI and generation of FFI bindings.
+//!
+//! Conversions:
+//! bool -> u8
+
+pub use iroha_ffi_derive::*;
+
+/// FFI compatible tuple with 2 elements
+#[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Pair<K, V>(pub K, pub V);
+
+/// Result of execution of an FFI function
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+// TODO: What should be the repr?
+#[repr(C)]
```

Comment:
For now the default: `usize` is fine. In the future, I believe that we should expand the error returns to include a bit more information.

Reply:
I'll be using `repr(i32)` at this point. I left a comment, we can revise it in the future